lib/saluki-env/src/workload/aggregator.rs

```diff
@@ -107,6 +126,22 @@ impl MemoryBounds for MetadataAggregator {
             // Operations channel.
             .with_array::<MetadataOperation>(OPERATIONS_CHANNEL_SIZE);
 
-        builder.with_subcomponent("tag_store", &self.tag_store);
+        for store in &self.borrowing_stores {
+            builder.with_subcomponent(store.store_name(), store);
+        }
+
+        for store in &self.consuming_stores {
+            builder.with_subcomponent(store.store_name(), store);
+        }
     }
 }
+
+pub trait ConsumingStore: MemoryBounds {
+    fn store_name(&self) -> &'static str;
+    fn process_operation(&mut self, operation: MetadataOperation);
+}
+
+pub trait BorrowingStore: MemoryBounds {
+    fn store_name(&self) -> &'static str;
+    fn process_operation(&mut self, operation: &MetadataOperation);
+}
```

Comment:
I know this is intended for use in an upcoming change, so this might be totally off the mark, but my initial reaction is skepticism that it's worth the conceptual overhead to have two different traits for this. I know it enables us to avoid extra cloning in some circumstances, but it's not obvious to me that we'll see actual benefit from that. Happy to wait and see though.

Reply:
Also a fair point.

I can collapse them for now and have a consuming variant only. I know that the upcoming usage _does_ benefit from only needing immutable access, but this codepath is fairly low volume. All of these optimizations are mostly just trying to preemptively stymie the growth of very small allocations over time... but maybe we can approach it differently in the future and do so in a lightweight way where we're not having to futz with the traits _here_, but with a global optimization somewhere else, etc.

I'll collapse them for now.

Reply:
Alright, I collapsed them for now, so we just have `MetadataStore` which is a consuming variant.